### PR TITLE
Generate and install a pkg-config file.

### PR DESCRIPTION
**Pull request by tlsa**
This adds a pkg-config file for libutf8proc.

Fixes issue https://github.com/JuliaStrings/utf8proc/issues/105.

### CODE REVIEW

**Comment by tlsa** [comment]
@stevengj Does this look reasonable?

File: Makefile [hunk 1]
```diff
@@ -34,8 +34,9 @@ endif
 
 # installation directories (for 'make install')
 prefix=/usr/local
-libdir=$(prefix)/lib
-includedir=$(prefix)/include
+libdir=lib
```
Comment (stevengj) [inline Makefile]: In the [GNU-style installation variables](https://www.gnu.org/prep/standards/html_node/Directory-Variables.html), `libdir` should be the whole path, defaulting to `$(prefix)/lib`.   So the previous version of this code was correct.

**Comment by tlsa** [comment]
@stevengj Oh, good point.  I still think it's worth ensuring the pkgconfig files stay in sync though, so I've reverted the previous change, and done it like this https://github.com/JuliaStrings/utf8proc/pull/142/commits/53fb330c59c7f1baaa49a10b2228f96bd7b80e6f instead.

File: Makefile [hunk 2]
```diff
@@ -36,6 +36,10 @@ endif
 prefix=/usr/local
 libdir=$(prefix)/lib
 includedir=$(prefix)/include
+pkgconfigdir=$(libdir)/pkgconfig
+
+pkglibdir=$(libdir:$(prefix)/%=%)
```
Comment (nalimilan) [inline Makefile]: I don't understand why you compute these variables. Why not use `libdir` directly in the pkgconfig file, and hardcode the prefix in each variable? I don't see the advantage of using `${prefix}` there.

Comment (tlsa) [reply Makefile]: Here are some examples:

* https://people.freedesktop.org/~dbn/pkg-config-guide.html
* https://linux.die.net/man/1/pkg-config
* https://en.wikipedia.org/wiki/Pkg-config

Comment (nalimilan) [reply Makefile]: I see that, but what's the point when it makes the code more complex?

Comment (tlsa) [reply Makefile]: When you run pkg-config, e.g. to get the appropriate cflags for compiling the library, you can use stuff like `pkg-config --define-prefix --cflags libpng`, which will change what is substituted for `$(prefix)` when it spits out the includedir.

Comment (nalimilan) [reply Makefile]: OK, makes sense then!

**Comment by tlsa** [comment]
@stevengj Is this OK now?

**Pull request merged**